src/Api/DataProvider/AbstractElasticSearchDataProvider.php

```diff
@@ -42,18 +42,18 @@ abstract class AbstractElasticSearchDataProvider
      *   Elastica index search type
      * @param RequestStack $requestStack
      *   Symfony request stack
-     * @param LoggerInterface $statsLogger
-     *   Logger for statistics
+     * @param StatsLoggingService $statsLoggingService
+     *   Statistics logging service
      * @param EventDispatcherInterface $dispatcher
      *   Symfony Event Dispatcher
      * @param IdentifierFactory $factory
      *   Factory to create Identifier Data Transfer Objects (DTOs)
      */
-    public function __construct(Type $index, RequestStack $requestStack, LoggerInterface $statsLogger, EventDispatcherInterface $dispatcher, IdentifierFactory $factory)
+    public function __construct(Type $index, RequestStack $requestStack, StatsLoggingService $statsLoggingService, EventDispatcherInterface $dispatcher, IdentifierFactory $factory)
```

Comment:
The logger service implements the `LoggerInterface` this should be the type that the function takes.

Reply:
But this is autowired. Therefore it needs to be the precise service.